frontera/contrib/middlewares/fingerprint.py

```diff
@@ -112,3 +113,19 @@ def _add_fingerprint(self, obj):
             for domain in obj.meta['redirect_domains']:
                 domain['fingerprint'] = self.fingerprint_function(domain['name'])
         return obj
+
+
+class RequestFingerprintMiddleware(BaseFingerprintMiddleware):
+
+    component_name = 'Request Fingerprint Middleware'
+    fingerprint_function_name = 'REQUEST_FINGERPRINT_FUNCTION'
+
+    def _get_fingerprint(self, url):
+        return self.fingerprint_function(Request(url))
+
+    def _add_fingerprint(self, obj):
+        request = obj if isinstance(obj, Request) else obj.request
+        obj.meta['fingerprint'] = self.fingerprint_function(request)
+        if 'redirect_urls' in obj.meta:
+            obj.meta['redirect_fingerprints'] = [self._get_fingerprint(url) for url in obj.meta['redirect_urls']]
```

Comment:
redirect URL fingerprints are already set in the meta, maybe above fingerprint calculations in SQLAlchemyBackend are not required.